app/services/ui_notifications.rb

```diff
@@ -0,0 +1,31 @@
+# Namespace for Foreman UI notifications event handling
+module UiNotifications
+  class Base
+    attr_reader :state, :subject
+
+    def initialize(subject, state)
+      raise(Foreman::Exception, 'must provide notification subject') if subject.nil?
```

Comment:
Please translate the string and use `::Foreman::Exception.new(N_("Some error"))` 

Nitpick but the string first letter should be uppercase